DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -929,3 +949,48 @@ int muon::sharedSegments( const reco::Muon& mu, const reco::Muon& mu2, unsigned
   
     return ret; 
 }
+
+void muon::setCutBasedSelectorFlags(reco::Muon& muon, 
+				    const reco::Vertex* vertex,
+				    bool run2016_hip_mitigation)
+{
+  // https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuideMuonIdRun2
+  unsigned int selection = muon.getSelectionMask();
+  // Compute Id and Isolation variables
+  double chIso  = muon.pfIsolationR04().sumChargedHadronPt;
+  double nIso   = muon.pfIsolationR04().sumNeutralHadronEt;
+  double phoIso = muon.pfIsolationR04().sumPhotonEt;
+  double puIso  = muon.pfIsolationR04().sumPUPt;
+  double dbCorrectedIsolation = chIso + std::max( nIso + phoIso - .5*puIso, 0. ) ;
+  double dbCorectedRelIso = dbCorrectedIsolation/muon.pt();
+  double tkRelIso = muon.isolationR03().sumPt/muon.pt();
+
+  // Base selectors
+  if (muon::isLooseMuon(muon))        selection |= reco::Muon::CutBasedIdLoose;
+  if (vertex){
+    if (muon::isTightMuon(muon,*vertex))  selection |= reco::Muon::CutBasedIdTight;
+    if (muon::isSoftMuon(muon,*vertex,run2016_hip_mitigation))   selection |= reco::Muon::SoftCutBasedId;
+    if (muon::isHighPtMuon(muon,*vertex)) selection |= reco::Muon::CutBasedIdGlobalHighPt;
+    if (muon::isTrackerHighPtMuon(muon,*vertex)) selection |= reco::Muon::CutBasedIdTrkHighPt;
+  }
+  if (muon::isMediumMuon(muon,run2016_hip_mitigation)){
+    selection |= reco::Muon::CutBasedIdMedium;
+    if ( vertex and 
+	 fabs(muon.muonBestTrack()->dz( vertex->position()))<0.1 and 
+	 fabs(muon.muonBestTrack()->dxy(vertex->position()))< 0.02 )
+      selection |= reco::Muon::CutBasedIdMediumPrompt;
+  }
+
+  // PF isolation
+  if (dbCorectedRelIso<0.40)    selection |= reco::Muon::PFIsoVeryLoose;
+  if (dbCorectedRelIso<0.25)    selection |= reco::Muon::PFIsoLoose;
+  if (dbCorectedRelIso<0.20)    selection |= reco::Muon::PFIsoMedium;
+  if (dbCorectedRelIso<0.15)    selection |= reco::Muon::PFIsoTight;
+  if (dbCorectedRelIso<0.10)    selection |= reco::Muon::PFIsoVeryTight;
```

Comment:
these magic constants (0.4, 0.25 etc) should be made consts of e.g. reco::Muon:: or in some other place and used consistently instead of entering values explicitly in multiple places and producers

Reply:
They are used in one place only and that place sets the flags. Where do you see them used beside that place?

Reply:
0.10 as a "Tight" appears multiple times in this PR, different variants of "Tight" though

Reply:
0.10 doesn't mean tight, it's just a number that was selected for different isolations that happens to represent tight. Nothing magical here.

Reply:
OK